### ISSUE TITLE
Popup not closing when in Menu

### ISSUE THREAD

**Issue by johannao76** [microsoft/fluent-ui-react#557]
### Steps
When using Popup with Menu, clicking on a menu item when another menu item's popup is open, will not close the original item. Use this code snippet in the docs site:
```jsx
import React from 'react'
import { Menu, Popup} from '@stardust-ui/react'

const items = [
  { key: 'editorials', content: 'Editorials' },
  { key: 'review', content: 'Reviews' },
  { key: 'events', content: 'Upcoming Events' },
]

const MenuExample = () => <Menu defaultActiveIndex={0} items={items} renderItem={renderItem}/>

const renderItem = (MenuItem, props) => {
  return (
    <Popup 
      trigger={<MenuItem {...props} />}
      content={{content: "TestContent"}}
    />
  )
}

export default MenuExample
```

### Expected Result
When a menu item's popup is open, clicking on another menu item should close the popup and open the clicked item's popup.

### Actual Result
We end up with two open popups

### Version
0.13.1

**Comment by miroslavstastny** [comment]
The problem is that when a `MenuItem` is clicked:
1. whole menu is updated,
2. this results in `Popup.componentDidUpdate()`,
3. which removes click listener.
4. and re-adds it in `setTimeout()`.

Document click event is called between steps 3 and 4 resulting in the event being missed by the `Popup`.

Click on the *Upcoming Events* opens popup:
```log
MenuItem.tsx:74 MenuItem::componentDidUpdate Upcoming Events
12:43:08.398 MenuItem.tsx:60 MenuItem::handleClick Upcoming Events
12:43:08.415 MenuItem.tsx:74 MenuItem::componentDidUpdate Editorials
12:43:08.416 Popup.tsx:104 Popup::updateSubcription Editorials
12:43:08.416 MenuItem.tsx:74 MenuItem::componentDidUpdate Reviews
12:43:08.417 Popup.tsx:104 Popup::updateSubcription Reviews
12:43:08.417 MenuItem.tsx:74 MenuItem::componentDidUpdate Upcoming Events
12:43:08.418 Popup.tsx:104 Popup::updateSubcription Upcoming Events
12:43:08.419 Menu.tsx:84 Menu::componentDidUpdate
12:43:08.425 EventTarget.tsx:16 _emit [ƒ]
12:43:08.425 Menu.tsx:80  Document.onClick
12:43:08.443 Popup.tsx:110 Popup - subscribe Upcoming Events
```

Click on the *Reviews*:
```log
12:43:14.041 MenuItem.tsx:74 MenuItem::componentDidUpdate Upcoming Events
12:43:14.045 MenuItem.tsx:74 MenuItem::componentDidUpdate Reviews
12:43:14.128 MenuItem.tsx:60 MenuItem::handleClick Reviews
12:43:14.141 MenuItem.tsx:74 MenuItem::componentDidUpdate Editorials
12:43:14.141 Popup.tsx:104 Popup::updateSubcription Editorials
12:43:14.142 MenuItem.tsx:74 MenuItem::componentDidUpdate Reviews
12:43:14.143 Popup.tsx:104 Popup::updateSubcription Reviews
12:43:14.143 MenuItem.tsx:74 MenuItem::componentDidUpdate Upcoming Events
12:43:14.144 Popup.tsx:104 Popup::unsubscribe Upcoming Events
12:43:14.144 Menu.tsx:84 Menu::componentDidUpdate
12:43:14.148 EventTarget.tsx:16 _emit [ƒ]
12:43:14.148 Menu.tsx:80  Document.onClick
12:43:14.154 Popup.tsx:110 Popup - subscribe Reviews
12:43:14.155 Popup.tsx:110 Popup - subscribe Upcoming Events
```

**Comment by kuzhelov** [comment]
addressed in #598
